packages/runtime-core/src/componentOptions.ts

```diff
@@ -578,6 +610,7 @@ export function applyOptions(
       deferredData.forEach(dataFn => resolveData(instance, dataFn, publicThis))
     }
     if (dataOptions) {
+      // @ts-ignore dataOptions is not fully type safe
```

Comment:
dataOptions get an invalid type here, I suspect is because the type became more strict